core/src/main/java/org/apache/calcite/plan/expand/ExpandJoinOnLeftCalcToJoinUnifyRule.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.plan.expand;
+
+import org.apache.calcite.plan.SubstitutionVisitor;
+import org.apache.calcite.rel.core.JoinRelType;
+import org.apache.calcite.rel.mutable.MutableCalc;
+import org.apache.calcite.rel.mutable.MutableJoin;
+import org.apache.calcite.rel.mutable.MutableRel;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexProgram;
+import org.apache.calcite.rex.RexShuttle;
+import org.apache.calcite.util.Pair;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.calcite.plan.SubstitutionVisitor.explainCalc;
+import static org.apache.calcite.plan.SubstitutionVisitor.splitFilter;
+import static org.apache.calcite.plan.expand.JoinRexUtils.addJoinDerivePredicate;
+import static org.apache.calcite.plan.expand.JoinRexUtils.extractJoinContext;
+import static org.apache.calcite.plan.expand.JoinRexUtils.isJoinRewritable;
+import static org.apache.calcite.plan.expand.JoinRexUtils.isSupportedPredict;
+import static org.apache.calcite.plan.expand.JoinRexUtils.tryCastRexInputRef;
+import static org.apache.calcite.plan.expand.SubstitutionRuleUtils.canPullUpFilterUnderJoin;
+import static org.apache.calcite.plan.expand.SubstitutionRuleUtils.fieldCnt;
+import static org.apache.calcite.plan.expand.SubstitutionRuleUtils.referenceByMapping;
+import static org.apache.calcite.plan.expand.SubstitutionRuleUtils.tryMergeParentCalcAndGenResult;
+
+/**
+ * Enhancements to joinOnLeftCalcToJoinUnifyRule.
```

Comment:
Can you make the comments more detailed? It's better to include examples in the comments so that others can understand them easily,the other classes are the same